### ISSUE TITLE
Add a tryMulticall function that allows individual calls to revert

### ISSUE THREAD

**Issue by bbenligiray** [api3dao/airnode-protocol-v1#19]
https://github.com/api3dao/airseeker/issues/216

Refer to https://github.com/makerdao/multicall/blob/master/src/Multicall2.sol#L55

**Comment by acenolaza** [comment]
@bbenligiray I remember that some time ago, Ashar needed to call some DapiServer view function multiple times. So what if we also add a view multicall function to the ExtendMulticall? I was thinking that maybe this function could use the low level `staticcall` instead of `delegatecall` when calling the view function (like `readDataFeedWithId`) but I think `staticcall` changes the `msg.sender` 🤔

**Comment by bbenligiray** [comment]
The current Multicall allows you to do batch static calls

**Comment by acenolaza** [comment]
Yes, if you do them offchain, right? But what if a `view` function in another contract wants to call `DapiServer.multicall()`? Wouldn't that force the calling function not be `view`? I saw that the openzeppelin Address library contains a [funcitonStaticCall()](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Address.sol#L145) function and ethereum docs say that any function in a library shares the same context as the calling contract so it would be a delegated call.

**Comment by bbenligiray** [comment]
> Wouldn't that force the calling function not be view?

Is this a problem?

**Comment by acenolaza** [comment]
> > Wouldn't that force the calling function not be view?
> 
> Is this a problem?

Not currently I guess 😬 I'm just trying to think ahead to avoid future DapiServer changes

**Comment by bbenligiray** [comment]
I mean is it ever a problem for a view function to not be declared as view. As far as I know, view doesn't change how the function behaves, it just has the compiler throw if the function does things that disables it from using the view modifier.

**Comment by acenolaza** [comment]
While that's true if the caller function is not defined as view then anyone calling it will need to submit a tx. I know that they can call it offchain statically but what if someone wanted to call that function from etherscan for instance? It the function was declared as view then it would be listed in the Read tab.

**Comment by bbenligiray** [comment]
Yeah, that's a good argument

**Comment by bbenligiray** [comment]
@acenolaza btw I changed my mind about this being a good argument, why would you ever want to do a multicall on Etherscan :smile:

**Comment by acenolaza** [comment]
Yeah, that's a good argument 🤣 

Let me keep thinking to see if I can come up with a better example/use case. Otherwise we can definitely drop the idea.
